### ISSUE TITLE
Compilation fails if storyboard name matches View Controller name 

### ISSUE THREAD

**Issue by vkaramov** [SwiftGen/SwiftGen#462]
If storyboard name matches the view controller name SwiftGen generated code looks like:

```
  internal enum ProfileSwitcherCVC: StoryboardType {
    internal static let storyboardName = "ProfileSwitcherCVC"

    internal static let initialScene = InitialSceneType<ProfileSwitcherCVC>(storyboard: ProfileSwitcherCVC.self)
  }
```

**Comment by AliSoftware** [comment]
Hi @vkaramov 

* Which version of SwiftGen are you using?
* Which template are you using to generate this code
* Which additional parameter did you pass to that template, if any?

It seems you either use an old version of SwiftGen, or that you passed the `ignoreTargetModule: true` parameter to the template when calling `swiftgen`. Because unless you set `ignoreTargetModule`, then the generated output should look like this, but instead should prefix the name of the class with the name of the module, exactly to avoid this kind of issue (we fixed that issue a while ago)

Here's how the code generated by the latest version of SwiftGen looks like by default: [see here an extract of our unit test suite results](https://github.com/SwiftGen/SwiftGen/blob/master/Tests/Fixtures/Generated/IB-iOS/swift4-context-all-ignore-module-need-extra-definitions.swift#L63)

```
    internal static let initialScene = InitialSceneType<LocationPicker.LocationPickerViewController>(storyboard: AdditionalImport.self)
```

As you can see it generates `LocationPicker.LocationPickerViewController` and not just `LocationPickerViewController`, avoiding the ambiguity. That issue has been solved a couple of versions back.

Please double-check all those questions we asked when you create a new issue (version of SwiftGen you're using, template you're using, invocation of `swiftgen` and/or config file you're using, parameters, you're passing)

**Comment by djbe** [comment]
If you're using your own template, then it's easy, just append "Storyboard" to the storyboard enum:
```swift
  internal enum ProfileSwitcherCVCStoryboard: StoryboardType {
    internal static let storyboardName = "ProfileSwitcherCVC"

    internal static let initialScene = InitialSceneType<ProfileSwitcherCVC>(storyboard: ProfileSwitcherCVCStoryboard.self)
  }
```

**Comment by djbe** [comment]
Closing this due to inactivity.

Note that we recently merged a fix for module parsing (#485), and if that doesn't fit your needs, a custom template may still be the best solution.
